refactor: separate k8s namespace from flow name (#3856)
jina-ai/jina: docs/advanced/experimental/kubernetes.md

All Executors in the Flow should be used with `jinahub+docker://...` or `docker://...`.
```

```{admonition} Caution
:class: caution

> We heavily recommend you to deploy `Flow` in a separate K8s namespace by passing `k8s_namespace` into the `Flow` arguments.

Reviewer: Actually that is not what I meant, sorry. I mean the user should not deploy it to an already existing namespace, which is used for something else like lets say 'very_valueable_production_services' ;)

embedding size:  512                                                                                       
```

You might have noticed that the above `Flow` has been deployed to the Namespace `example-clip`.

> By default, we use the `Flow` name as the `Kubernetes` namespace.

Reviewer: I would add a sentence that we heavily recommend to deploy ` Flow` in a separate K8s namespace. It does not have to have the Flow's name, but it probably should not run other services unless the user really knows what they are doing. We are doing a lot of things in the background in the cluster, like changing security permissions.

Author: added to doc top

> However, you can deploy `Flow` to a given `Namespace` you want, such as:

Reviewer: ```suggestion
However, you can deploy `Flow` to any given `Namespace`, like this:
```

> However, you can deploy `Flow` to a given `Namspace` you wanted, such as:

Reviewer 2: ```suggestion
However, you can deploy `Flow` to a given `Namespace` you want, such as:
```

embedding size:  512                                                                                       
```

You might have noticed that the above `Flow` has been deployed to the Namespace `example-clip`.
By default, we use the `Flow` name as the `Kubernetes` namespace.
However, you can deploy `Flow` to any given `Namespace`, like this:

```{code-block} python
---
emphasize-lines: 8
---
from jina import Flow

f = Flow(
    name='example-clip',
    port_expose=8080,
    infrastructure='K8S',
    protocol='http',
    k8s_namespace='custom-namespace',
).add(uses='jinahub+docker://CLIPImageEncoder')
```

```{admonition} Caution
:class: caution

> We heavily recommend you to deploy different `Flow` into separated namespaces. To be more concrete,

Reviewer: Maybe like this? I want to emphasize both that each Flow ideally is separated and in particular should not be mixed with services outside of Jina. Those should live in other namespaces.

```suggestion
We heavily recommend you to deploy each `Flow` into a separate namespace. In particular it should not be deployed into namespaces, where other essential non Jina services are running. 
```
